napi_memory_management.md

```diff
@@ -0,0 +1,143 @@
+# Memory Management
```

Comment:
Should this be titled something like "Object lifetime management" instead?

The only APIs we have that are directly related to memory-management are the finalize callbacks for `napi_wrap` and `napi_create_external` that enable freeing an external pointer, but those are not even covered here. (They'll be in a separate topic about object wrapping.)